lib/workload/stateful/filemanager/filemanager/src/events/aws/mod.rs

```diff
@@ -144,11 +140,62 @@ impl From<FlatS3EventMessages> for TransposedS3EventMessages {
     }
 }
 
+impl From<TransposedS3EventMessages> for FlatS3EventMessages {
```

Comment:
This definitely needs a comment, I always forget why the flattening and transposing is needed :-S

Reply:
Yes, fair. The transpose is just to help the shape of the query entering the database. I.e. using `unnest` on the input parameters.